src/Form/PeriodForm.php

```diff
@@ -119,6 +120,27 @@ public function form(FormBuilderInterface $builder, array $options) : void
                 $days->setData(array_fill(0, 7, []));
             }
         });
+
+        // Reset days info language to match the selected language.
+        $builder->addEventListener(FormEvents::SUBMIT, function (FormEvent $event) {
+            $period = $event->getData();
+            $form = $event->getForm();
+            $temp_lang = SystemLanguages::TEMPORARY_LANGCODE;
+
+            if ($period->isNew() && $form->has('langcode')) {
+                $days = $period->getDays();
+                $langcode = $form->get('langcode')->getData();
+
+                foreach ($days as $key => $day) {
+                    if(array_key_exists($temp_lang, $day['info'])) {
```

Comment:
Might be useful to comment here that individual days can have 'xx': null set and that needs to be replaced with the correct language of the period.

Reply:
I added a additional comment to that spot.